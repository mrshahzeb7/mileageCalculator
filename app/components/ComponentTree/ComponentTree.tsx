import React from 'react';
import {RowInput, RowInputProps} from '../RowInput/RowInput';
import {StyleProp, View, ViewStyle} from 'react-native';
import {spacing} from '../../theme';

interface Props {
  list: RowInputProps[];
  style: StyleProp<ViewStyle>;
}

export const ComponentTree: React.FC<Props> = ({style, list}) => {
  return (
    <View style={style}>
      {list?.map?.((item: RowInputProps, index: number) => {
        return (
          <RowInput
            style={$container}
            key={index}
            icon={item.icon}
            ChildOne={item.ChildOne}
            ChildTwo={item.ChildTwo}
          />
        );
      })}
    </View>
  );
};

const $container: ViewStyle = {
  marginTop: spacing.vertical.small,
  alignSelf: 'center',
  width: '100%',
  justifyContent: 'space-between',
};
