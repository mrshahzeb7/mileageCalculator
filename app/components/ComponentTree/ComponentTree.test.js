import 'react-native';
import React from 'react';
import {ComponentTree} from './ComponentTree';
import renderer from 'react-test-renderer';

describe('ComponentTree', () => {
  it(' ComponentTree renders correctly', () => {
    const tree = renderer.create(<ComponentTree />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
