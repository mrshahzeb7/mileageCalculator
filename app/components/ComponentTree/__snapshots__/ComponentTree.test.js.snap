// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentTree  ComponentTree renders correctly 1`] = `<View />`;
